components/file_upload/file_upload.jsx

```diff
@@ -351,7 +351,21 @@ export default class FileUpload extends PureComponent {
 
         this.props.onUploadError(null);
 
-        var files = e.originalEvent.dataTransfer.files;
+        const items = e.originalEvent.dataTransfer.items || [];
+        const droppedFiles = e.originalEvent.dataTransfer.files;
+        const files = [];
+        Array.from(droppedFiles).forEach((file, index) => {
+            const item = items[index];
+            if (item && item.webkitGetAsEntry && item.webkitGetAsEntry().isDirectory) {
+                return;
+            }
+            files.push(file);
+        });
+
+        if (files.length === 0) {
+            this.props.onUploadError(localizeMessage('file_upload.drag_folder', 'Folders cannot be uploaded. Please drag all files separately.'));
```

Comment:
Unfortunately, it's not that simple. If we were just running on Chrome or FF then that would be fine but the file length being zero is indicative of folders being dropped on both IE and Edge since the web API feature used in the loop above is not implemented for those browsers. The condition for the error isn't perfect, but it should be good enough for most cases.